test/e2e/usersignup_test.go

```diff
@@ -46,36 +53,482 @@ func TestUserSignup(t *testing.T) {
 	t.Logf("user signup '%s' created", userSignup.Name)
 
 	// Confirm the UserSignup was created
-	err = verifyResources(t, f, namespace, userSignup)
+	err = waitForUserSignup(t, f.Client.Client, userSignup.Name, namespace)
 	require.NoError(t, err)
 
+	// Confirm that a MasterUserRecord wasn't created
+	err = waitForMasterUserRecord(t, f.Client.Client, userSignup.Name, namespace)
+	require.Error(t, err)
+
 	// Delete the User Signup
+	err = f.Client.Delete(context.TODO(), userSignup)
+	require.NoError(t, err)
+}
+
+func TestUserSignupWithNoApprovalConfig(t *testing.T) {
+	// Test Setup
+	userSignupList := &v1alpha1.UserSignupList{}
+
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, userSignupList)
+	require.NoError(t, err, "failed to add custom resource scheme to framework")
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	err = ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err, "failed to initialize cluster resources")
+	t.Log("Initialized cluster resources")
+
+	namespace, err := ctx.GetNamespace()
+	require.NoError(t, err, "failed to get namespace where operator needs to run")
+
+	f := framework.Global
+
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "host-operator", 1, operatorRetryInterval, operatorTimeout)
```

Comment:
You could avoid a lot of code duplication (initializing cluster & waiting for deployment, etc) by using test suites or subtests.

Reply:
now using a test suite.